Automattic/wp-calypso
# Subscribers: Update Odyssey stats links to JP Cloud

Related to https://github.com/Automattic/dotcom-forge/issues/6364

## Proposed Changes

Update `View details` / `View all` link on Stats > Subscribers (tab) to Jetpack Cloud Subscribers for Atomic & Self hosted.
This also should update on Calypso and wp-admin Odyssey stats. 

![image](https://github.com/Automattic/wp-calypso/assets/402286/c83eb339-df4b-453c-9061-db56723e5237)

## Testing Instructions

* Go to `/apps/odyssey-stats`
* Run `NODE_ENV=production yarn dev --sync` to sync the changes to your sandbox
* Sandbox `widgets.wp.com`
* Use a site with >0 subscribers
* On Calypso, go to `/stats/subscribers/:SITE_SLUG`
* On wp-admin, go to `/wp-admin/admin.php?page=stats#!/stats/subscribers/:SITE_SLUG`
* The `View details` at the bottom of the page should link to Jetpack Cloud subscribers
